File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        if len(arr)<k:
            return 0
        count=0
        total_k=sum(arr[:k])
        if total_k/k>=threshold:
            count+=1


        left=1
        right=left+k-1
        
        while right<len(arr):
            print(total_k)
            total_k=total_k-arr[left-1]+arr[right]
            if total_k/k>=threshold:
                count+=1
            left+=1
            right+=1
        return count
    
    
        

       
                
            
            
        return count

        

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int left=0;
        int right=left+k-1;

        int total=0;
        int count=0;

        for( int i=0;i<k;i++){
            total+=arr[i];
        }

        while (right<arr.length){
            if(left>0){
                total=total-arr[left-1]+arr[right];
            }
            if(total/k>=threshold){
                count++;
            }
            left++;
            right++;
        }
        return count;
    }
}